ui/src/components/Tiles.js

```diff
@@ -0,0 +1,69 @@
+import React, { PropTypes, Component } from 'react';
+
+export default class Tiles extends Component {
+  render() {
+    var rows = [];
+    var tilesType = this.props.tilesType;
+    this.props.tiles.forEach(function(webtile) {
+      var thumbStyle = {
+        backgroundImage: 'url(' + webtile.imageURI + ')'
+      };
+      var enhancedThumbStyle = {
+        backgroundImage: 'url(' + webtile.enhancedImageURI + ')'
+      };
+      var frecent = [];
+      if (tilesType === 'suggested') {
+        webtile.frecent_sites.forEach(function(url){
+          frecent.push(
+            <li>{url}</li>
+          );
+        });
+      }
+      rows.push(
+        <div key={webtile.directoryId} className="newtab-cell">
+          <div className={'newtab-site ' + tilesType} type={webtile.type}>
+            <a className="newtab-link" href={webtile.url} title={webtile.title}>
+              <span className="newtab-thumbnail" style={thumbStyle}></span>
+              <span className="newtab-thumbnail enhanced-content" style={enhancedThumbStyle}></span>
+              <span className="newtab-title">{webtile.title}</span>
+            </a>
+            <span className="newtab-suggested">
+              <span className="newtab-suggested-bounds">
+                {webtile.explanation}
+              </span>
+            </span>
+          </div>
+          {tilesType === 'suggested' &&
+            <div className="frecent-sites">
+              <strong>Frecent Sites:</strong>
+              <ul>
+                {frecent}
+              </ul>
+            </div>
+          }
+        </div>
+      );
+    });
+
+    return (
+      <div id="newtab-grid">
+        {rows}
+      </div>
+    );
+  }
+}
+
+Tiles.propTypes = {
+  tiles: PropTypes.arrayOf(
+    PropTypes.shape({
+      title: PropTypes.string.isRequired,
+      imageURI: PropTypes.string.isRequired,
+      enhancedImageURI: PropTypes.string,
+      type: PropTypes.string,
```

Comment:
Made type required: 7b30fa7742c16b9264e66f20ae3c8902c7fb1305
Added adgroup_categories and frequency_caps: ff4c8fc9ba77f6b1146247dc028760d4c00e4b85